src/main/java/datavalidation/DataValidatorImpl.java

```diff
@@ -0,0 +1,38 @@
+package datavalidation;
+
+import fruitsassortment.ListOfFruits;
+import shopoperations.ListOfOperations;
+
+public class DataValidatorImpl implements DataValidator {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int PRODUCT_NAME_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public boolean validateData(String data) {
+        if ("type,fruit,quantity".equals(data.trim())) {
+            return true;
+        }
+        String dataType = data.split(SEPARATOR)[OPERATION_TYPE_INDEX].toUpperCase().trim();
+        if (!ListOfOperations.contains(dataType)) {
+            throw new RuntimeException("Such operation is not valid "
+                    + dataType);
+        }
+        dataType = data.split(SEPARATOR)[PRODUCT_NAME_INDEX].trim();
+        if (!ListOfFruits.contains(dataType.toUpperCase())) {
```

Comment:
I wouldn't hardcode list of fruits in the code, if you need to change the code each time you add / remove some fruit someone always needs to maintain the code. In future it will be handled by DB but for now I would remove this check